### PR TITLE
Update haproxy to latest LTS

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Upgrades haproxy from an unmaintained version to the most recent LTS including a recent security patch.

medic/cht-core#8076

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8076-update-haproxy/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8076-update-haproxy/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8076-update-haproxy/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: haproxy/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
-FROM haproxy:2.3.19
+FROM haproxy:2.6
```
Comment (garethbowen) [inline haproxy/Dockerfile:1]: I've chosen to not lock down the service pack version. This introduces a change of random build failures but means that we keep up to date with patches without any work. Let me know how you feel about this tradeoff.

File: haproxy/Dockerfile [hunk 2]
```diff
@@ -10,4 +11,4 @@ ADD rsyslog.conf /etc/rsyslog.conf
 COPY scripts /usr/local/etc/haproxy/
 
 ENTRYPOINT ["/entrypoint.sh"]
-CMD ["-f", "/usr/local/etc/haproxy/default_frontend.cfg", "-f", "/usr/local/etc/haproxy/backend.cfg"]
+CMD ["haproxy", "-f", "/usr/local/etc/haproxy/default_frontend.cfg", "-f", "/usr/local/etc/haproxy/backend.cfg"]
```
Comment (garethbowen) [inline haproxy/Dockerfile:14]: New change to use the haproxy user instead of root.

File: haproxy/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,6 @@
-FROM haproxy:2.3.19
+FROM haproxy:2.6
 
+USER root
```
Comment (garethbowen) [inline haproxy/Dockerfile:3]: Needed when going from 2.3 -> 2.4

**Comment by garethbowen** [comment]
Code review please @henokgetachew !

File: haproxy/Dockerfile (hunk 2 above)
Comment (henokgetachew) [reply haproxy/Dockerfile:14]: I'm not completely sure about this but don't you'd have to specify the user in the `cfg` file for haproxy to use that user? I think you'd also add it in the Dockerfile using `RUN useradd` to create the user and give it permissions to handle the haproxy files etc.

**Review by henokgetachew — CHANGES_REQUESTED** [review]
Cool change. I've left a comment about the haproxy user.

File: haproxy/Dockerfile (hunk 2 above)
Comment (henokgetachew) [reply haproxy/Dockerfile:14]: nvm. I just checked to confirm and I see the base image already does this.

**Review by henokgetachew — APPROVED** [review]
LGTM.

**Pull request merged**